src/core/runtime.d

```diff
@@ -494,32 +620,60 @@ extern (C) bool runModuleUnitTests()
         }
     }
 
-    if( Runtime.sm_moduleUnitTester is null )
+    if (Runtime.sm_extModuleUnitTester !is null)
+        return Runtime.sm_extModuleUnitTester();
+    else if (Runtime.sm_moduleUnitTester !is null)
+        return Runtime.sm_moduleUnitTester() ? UnitTestResult.pass : UnitTestResult.fail;
+    UnitTestResult results;
+    foreach( m; ModuleInfo )
     {
-        size_t failed = 0;
-        foreach( m; ModuleInfo )
+        if( m )
         {
-            if( m )
-            {
-                auto fp = m.unitTest;
+            auto fp = m.unitTest;
 
-                if( fp )
+            if( fp )
+            {
+                ++results.executed;
+                try
                 {
-                    try
-                    {
-                        fp();
-                    }
-                    catch( Throwable e )
-                    {
-                        _d_print_throwable(e);
-                        failed++;
-                    }
+                    fp();
+                    ++results.passed;
+                }
+                catch( Throwable e )
+                {
+                    _d_print_throwable(e);
                 }
             }
         }
-        return failed == 0;
     }
-    return Runtime.sm_moduleUnitTester();
+    import rt.config : rt_configOption;
+    if (results.passed != results.executed)
+    {
+        // by default, we always print a summary if there are failures.
+        results.summarize = true;
+    }
+    else switch (rt_configOption("testmode"))
+    {
+    case "":
```

Comment:
Shouldn't everything in this block be indented one level?

Reply:
Hm... I generally don't indent switch case labels. I'm not sure what the style guide says. I see switch statements elsewhere that don't have case labels indented: https://github.com/dlang/druntime/blob/master/src/core/time.d#L352

Reply:
Yeah, I was surprised now to see how many place in Phobos that don't indent cases in switch statements. I always indent the content of a block.